front/src/common/api/osrdEditoastApi.ts

```diff
@@ -3324,6 +3327,10 @@ export type SimulationSummaryResult =
         error_type: string;
       };
     };
+export type TrainScheduleForm = TrainScheduleBase & {
+  /** Timetable attached to the train schedule */
+  timetable_id?: number | null;
```

Comment:
timetable_id is optional/null to update a trainschedule but compulsory to post one. Is that normal ?

Reply:
Yes, a train schedule must be attached to a timetable. So you have to provide one at creation (it's provided in the url as path parameter).
To patch a train schedule if the user doesn't provide a timetable_id, then the train schedule stay attach to the same timetable.
This field has been added to allow the front end to move a schedule train (without having to delete and recreate it).

Reply:
ok got it !